core/vtk/ttkPlanarGraphLayout/ttkPlanarGraphLayout.cpp

```diff
@@ -144,3 +151,73 @@ int ttkPlanarGraphLayout::RequestData(vtkInformation *request,
 
   return 1;
 }
+
+template <class dataType>
+int ttkPlanarGraphLayout::mergeTreePlanarLayoutCallTemplate(
+  vtkUnstructuredGrid *treeNodes,
+  vtkUnstructuredGrid *treeArcs,
+  vtkUnstructuredGrid *output) {
+  int verbose = 0;
+  MergeTree mergeTree = makeTree<dataType>(treeNodes, treeArcs);
+  FTMTree_MT *tree = &(mergeTree.tree);
+  // tree->printTree2();
+
+  computePersistencePairs<dataType>(tree);
+
+  std::vector<std::vector<int>> treeNodeCorrMesh(1);
+  treeNodeCorrMesh[0] = std::vector<int>(tree->getNumberOfNodes());
+  for(unsigned int j = 0; j < tree->getNumberOfNodes(); ++j)
+    treeNodeCorrMesh[0][j] = j;
+
+  ttkMergeTreeVisualization visuMaker;
+  visuMaker.setPlanarLayout(true);
+  visuMaker.setOutputSegmentation(false);
+  visuMaker.setBranchDecompositionPlanarLayout(BranchDecompositionPlanarLayout);
+  visuMaker.setBranchSpacing(BranchSpacing);
+  visuMaker.setImportantPairs(ImportantPairs);
+  visuMaker.setImportantPairsSpacing(ImportantPairsSpacing);
+  visuMaker.setNonImportantPairsSpacing(NonImportantPairsSpacing);
+  visuMaker.setNonImportantPairsProximity(NonImportantPairsProximity);
+  visuMaker.setVtkOutputNode(output);
+  visuMaker.setVtkOutputArc(output);
+  visuMaker.setTreesNodes(treeNodes);
+  visuMaker.setTreesNodeCorrMesh(treeNodeCorrMesh);
+  visuMaker.makeTreesOutput<dataType>(tree, verbose);
+
+  return 1;
+}
+
+int ttkPlanarGraphLayout::mergeTreePlanarLayoutCall(
+  vtkInformation *request,
+  vtkInformationVector **inputVector,
+  vtkInformationVector *outputVector) {
+
+  vtkUnstructuredGrid *treeNodes
+    = vtkUnstructuredGrid::GetData(inputVector[0], 0);
+  vtkUnstructuredGrid *treeArcs
+    = vtkUnstructuredGrid::GetData(inputVector[0], 1);
+  auto output = vtkUnstructuredGrid::GetData(outputVector);
+  int dataTypeInt
```

Comment:
is this possible the "Scalar" array does not exist ?

Reply:
Hum, I don't think so, the planar graph layout (of this PR) should be executed with the nodes and arcs output of a FTMTree filter, and the nodes output has a "Scalar" array.